### PR TITLE
[WIP] Juicebox bridge

### PR DESCRIPTION
**Pull request by dhyon**
context: https://github.com/AztecProtocol/aztec-connect-bridges

Will work through the pay functionality first before doing the redeem functionality.

- [ ] Write a Solidity bridge that interfaces with the protocol you are bridging e.g AAVE
- [ ] Write tests in Solidity that test the bridge with production values and the deployed protocol that is currently on mainnet. You should test a range of assets and edge cases and use Forge's fuzzing abilities.
- [ ] Implement the typescript bridge-data.ts class that tells a front-end developer how to use your bridge. 
- [ ] Write an explanation of the flows your bridge supports to be included as spec.md
- [ ] Deploy your bridge

### CODE REVIEW

File: src/bridges/juicebox/JuiceboxBridge.sol [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            0,
+            /* _preferClaimedTokens */
+            true,
+            /* _memo */
+            'Batched payments from Aztec L2. Privacy preserved!',
+            /* _delegateMetadata */
+            new bytes(0)
+        );
+
+        // TODO: best way to get or predict outputValueA aka "number of resulting JBTokens from pay"?
```
Comment (mejango) [inline src/bridges/juicebox/JuiceboxBridge.sol:95]: current balanceOf outputAssetA - previous balanceOf outputAssetA

**Comment by PhilWindle** [comment]
Thanks for the work so far @dhyon. Were you planning on writing some tests to demonstrate the bridge working correctly? We are using foundry as the framework for writing and executing tests.

**Comment by dhyon** [comment]
> Thanks for the work so far @dhyon. Were you planning on writing some tests to demonstrate the bridge working correctly? We are using foundry as the framework for writing and executing tests.

Thanks @PhilWindle. Yeah the fun part of testing is up next. I'm wondering why the tests need to be written for prod (mainnet) values. Is it a hard requirement on your end? Reason I'm asking is because we're building this bridge for V2 of our protocol which currently is only deployed on Rinkeby atm.

**Comment by benesjan** [comment]
> > Thanks for the work so far @dhyon. Were you planning on writing some tests to demonstrate the bridge working correctly? We are using foundry as the framework for writing and executing tests.
> 
> Thanks @PhilWindle. Yeah the fun part of testing is up next. I'm wondering why the tests need to be written for prod (mainnet) values. Is it a hard requirement on your end? Reason I'm asking is because we're building this bridge for V2 of our protocol which currently is only deployed on Rinkeby atm for stage testing.

Hello, sorry for not getting any response for so long. We are testing against mainnet simply because all the other DeFi protocols are already deployed. Is this problem still relevant? If yes we could change the configuration for this bridge specifically.

**Comment by LHerskind** [comment]
Closed based on inactivity. Feel free to open when there is a completed bridge.
